### PR TITLE
tests: posix: eventfd testsuite improvements

### PR DESCRIPTION
**Pull request by cfriedt**
6 commits. No change in functionality. Mainly just reducing duplicate code and redundant churn in CI.

* tests: posix: eventfd: consilidate into one testsuite
* tests: posix: eventfd: consistent testsuite name
* tests: posix: eventfd: be less repetitive in naming
* tests: posix: eventfd: reduce testing requirements and scope
* tests: posix: eventfd: normalize includes
* tests: posix: eventfd: split into blocking and nonblocking

Note: there is a bug in eventfd, and this PR is really just getting the testsuite into shape so that it's easy to demonstrate the bug (as described in #58426).

### CODE REVIEW

**Comment by cfriedt** [comment]
For some reason, git thinks my user.name is "Christopher" when it's only set to "Chris". Bit of an annoyance... might need to fix tomorrow.

File: tests/posix/eventfd/src/blocking.c [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+	ret = poll(&pfd, 1, 500);
+	zassert_true(ret == 0, "poll ret %d", ret);
+}
+
+ZTEST_F(eventfd, test_set_poll_event_block)
+{
+	close(fixture->fd);
+
+	fixture->fd = eventfd(TESTVAL, 0);
```
[the remark this answers is not in the document]
Comment (cfriedt) [reply tests/posix/eventfd/src/blocking.c]: Fixed!

**Comment by cfriedt** [comment]
Minor annoyance on a different machine with git `user.name` being "Chris" instead of "Christopher". Would prefer the short version at some point, if I can fix that.

**Review by nashif — DISMISSED** [review]

**Review by ycsin — DISMISSED** [review]

File: tests/posix/eventfd/src/_main.c [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	fixture.fd = -1;
+	return &fixture;
+}
+
+static void before(void *arg)
+{
+	struct eventfd_fixture *fixture = arg;
+
+	fixture->fd = eventfd(0, 0);
+	zassert_true(fixture->fd >= 0, "eventfd(0, 0) failed: #d", errno);
```
Comment (cfriedt) [inline tests/posix/eventfd/src/_main.c:97]: ugh - typo 😞

**Comment by cfriedt** [comment]
@nashif, @ycsin - please re-ack (I noticed a typo in a print format specifier and needed to fix)

@jgl-meta - you should be OK to merge this once it is green since it's not adding any new features and just improving the testsuite.

**Review by ycsin — APPROVED** [review]

**Review by jgl-meta — APPROVED** [review]

**Pull request merged**